src/Tools/dotnet-monitor/Commands/CommandExtensions.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading;
+using System.Threading.Tasks;
+
+#if UNITTEST
+namespace Microsoft.Diagnostics.Monitoring.UnitTestApp.Scenarios
+#else
+namespace Microsoft.Diagnostics.Tools.Monitor.Commands
+#endif
+{
+    internal static class CommandExtensions
+    {
+        // Due to https://github.com/dotnet/command-line-api/pull/2095, returning an exit code is "pay-for-play". A custom CliAction
+        // must be implemented in order to actually observe the exit code. Otherwise, SetAction will happily accept a callback that
```

Comment:
This seems like a bug in commandline?

Reply:
https://github.com/dotnet/command-line-api/issues/2109